core/src/main/java/org/apache/carbondata/scan/result/vector/CarbonColumnVector.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.carbondata.scan.result.vector;
+
+import org.apache.spark.sql.types.Decimal;
+
+public interface CarbonColumnVector {
+
+  public void putShort(int rowId, short value);
+
+  public void putInt(int rowId, int value);
+
+  public void putLong(int rowId, long value);
+
+  public void putDecimal(int rowId, Decimal value, int precision);
+
+  public void putDouble(int rowId, double value);
+
+  public void putBytes(int rowId, byte[] value);
+
+  public void putBytes(int rowId, int offset, int length, byte[] value);
```

Comment:
Yes, but I mean this function `putBytes(int rowId, int offset, int length, byte[] value)` is never used